src/features/broker-protection/actions/extract.js

```diff
@@ -278,114 +274,46 @@ export function aggregateFields (profile) {
+ * @param {string} outputFieldKey
+ * @param {ExtractProfileProperty} extractorParams
+ * @param {string[]} elementValues
+ * @return {any}
  */
-export function extractValue (key, value, elementValue) {
-    if (!elementValue) return null
-
-    const extractors = {
-        name: () => typeof elementValue === 'string' && elementValue.replace(/\n/g, ' ').trim(),
-        age: () => typeof elementValue === 'string' && elementValue.match(/\d+/)?.[0],
-        alternativeNamesList: () => stringToList(elementValue, value.separator),
-        addressCityStateList: () => {
-            const cityStateList = stringToList(elementValue, value.separator)
-            return getCityStateCombos(cityStateList)
-        },
-        addressCityState: () => {
-            const cityStateList = stringToList(elementValue)
-            return getCityStateCombos(cityStateList)
-        },
-        addressFullList: () => stringToList(elementValue, value.separator),
-        phone: () => {
-            const phoneNumber = typeof elementValue === 'string' && elementValue.replace(/\D/g, '')
-            if (!phoneNumber) {
-                return null
-            }
-            return stringToList(phoneNumber)
-        },
-        addressFull: () => elementValue,
-        phoneList: () => stringToList(elementValue, value.separator),
-        relativesList: () => stringToList(elementValue, value.separator),
-        profileUrl: () => {
-            const profile = {
-                profileUrl: elementValue,
-                identifier: elementValue
-            }
-
-            if (!value.identifierType || !value.identifier) {
-                return profile
-            }
-
-            const profileUrl = Array.isArray(elementValue) ? elementValue[0] : elementValue
-            profile.identifier = getIdFromProfileUrl(profileUrl, value.identifierType, value.identifier)
-            return profile
-        }
+export function extractValue (outputFieldKey, extractorParams, elementValues) {
+    switch (outputFieldKey) {
+    case 'age': return new AgeExtractor().extract(elementValues, extractorParams)
+    case 'name': return new NameExtractor().extract(elementValues, extractorParams)
+
+    // all addresses are processed the same way
+    case 'addressFull':
+    case 'addressFullList':
+        return new AddressFullExtractor().extract(elementValues, extractorParams)
+    case 'addressCityState':
+    case 'addressCityStateList':
+        return new CityStateExtractor().extract(elementValues, extractorParams)
+
+    case 'alternativeNamesList': return new AlternativeNamesExtractor().extract(elementValues, extractorParams)
+    case 'relativesList': return new RelativesExtractor().extract(elementValues, extractorParams)
+    case 'phone':
+    case 'phoneList':
+        return new PhoneExtractor().extract(elementValues, extractorParams)
+    case 'profileUrl': return new ProfileUrlExtractor().extract(elementValues, extractorParams)
```

Comment:
Should we have a default case that throws a warning? We can leave that for another time if you'd like since we're not doing anything with warnings yet.

Reply:
yep, let's follow up